parser/parser_test.go

```diff
@@ -1541,6 +1542,52 @@ foo: > # comment
 # This comment is in its own document
 ---
 a: b
+`,
+		},
+		{
+			name: "map with misaligned indentation in comments",
+			yaml: `
+ # commentA
+a:  #commentB
+   # commentC
+  b: c  # commentD
+    # commentE
+  d: e  # commentF
+ # commentG
+`,
+			expected: `
+# commentA
+a: #commentB
+  # commentC
+  b: c # commentD
+  # commentE
+  d: e # commentF
+# commentG
+`,
+		},
+		{
+			name: "sequence with misaligned indentation in comments",
+			yaml: `
+ # commentA
+- a  # commentB
+ # commentC
+- b:   # commentD
+   # commentE
+  - d  # commentF
+    # commentG
+  - e  # commentG
+ # commentH
+`,
+			expected: `
+# commentA
```

Comment:
Ohh, I also overlooked that this test specifically makes it so that it's expected for this to modify comment indentation.

I'm not sure that this makes sense for the default behaviour. Since if comments are supposed to be parsed and respected the same as other fields, then changing indentation should be seen as breaking. However if someone wanted to "beautify" a file I could see this behaviour as beneficial via opt-in setting.